common/server_protocolhandler.cpp

```diff
@@ -542,6 +542,24 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
     if (!r)
         return Response::RespNameNotFound;
 
+    QString roomPermission = r->getRoomPermission().toLower();
+    if (roomPermission != "none"){
+        if (roomPermission == "registered") {
```

Comment:
Enum would be good, I agree. I think it's something we will likely revisit later so it's not such a big deal